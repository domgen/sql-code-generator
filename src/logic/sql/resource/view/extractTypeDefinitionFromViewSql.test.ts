describe('extractTypeDefinitionFromViewSql', () => {
  it('should be able to extract types in this example', async () => {
    // const exampleSql = await getSqlFromFile({ filePath: `${__dirname}/__test_assets__/image.sql` });
    // const typeDef = extractTypeDefinitionFromViewSql({ name: 'image', sql: exampleSql });
    // expect(typeDef).toMatchSnapshot();
  });
});
