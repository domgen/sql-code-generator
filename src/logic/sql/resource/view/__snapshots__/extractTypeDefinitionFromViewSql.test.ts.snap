// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractTypeDefinitionFromViewSql should be able to extract types in this example 1`] = `
TypeDefinitionOfResourceView {
  "name": "view_suggestion_current",
  "selectExpressions": Array [
    TypeDefinitionOfQuerySelectExpression {
      "alias": "id",
      "sourcePath": "s.id",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "uuid",
      "sourcePath": "s.uuid",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "suggestion_source",
      "sourcePath": "s.suggestion_source",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "external_id",
      "sourcePath": "s.external_id",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "status",
      "sourcePath": "v.status",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "result",
      "sourcePath": "v.result",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "created_at",
      "sourcePath": "s.created_at",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "effective_at",
      "sourcePath": "v.effective_at",
    },
    TypeDefinitionOfQuerySelectExpression {
      "alias": "updated_at",
      "sourcePath": "v.created_at",
    },
  ],
  "tableReferences": Array [
    TypeDefinitionOfQueryTableReference {
      "alias": "s",
      "tableName": "suggestion",
    },
    TypeDefinitionOfQueryTableReference {
      "alias": "c",
      "tableName": "suggestion_cvp",
    },
    TypeDefinitionOfQueryTableReference {
      "alias": "v",
      "tableName": "suggestion_version",
    },
  ],
}
`;
